### PR TITLE
Changes

### PR DESCRIPTION
**Pull request by KritantaDev**
WIP

### CODE REVIEW

**Comment by KritantaDev** [comment]
Much more to be done, still some issues with class names for some reason, but it's now in a much better state.

**Comment by KritantaDev** [comment]
Personal roadmap:

- Add a logging/debugging class and optionally log as much as possible
- Testing:
1. Find a dylib in a cache I have an officially extracted copy of
2. Set up testing against that cache to diff and find issues with extraction
- IDA Plugin
- Rework this project into a pip-compatible module

File: DyldExtractor/Converter/ObjCConvertor.py [hunk 1]
```diff
@@ -75,11 +75,22 @@ def convert(self) -> None:
 	def processSegments(self) -> None:
 		segments: List[MachO.segment_command_64] = self.machoFile.getLoadCommand(MachO.LoadCommands.LC_SEGMENT_64, multiple=True)
 
+		# Segments we need to zero out a byte on
```
Comment (arandomdev) [inline DyldExtractor/Converter/ObjCConvertor.py:78]: This should be unnecessary, the extra byte in the pointer is for rebase data. This should have been taken care of by using RebaseConverter.

File: DyldExtractor/Converter/SelectorConvertor.py [hunk 2]
```diff
@@ -74,7 +74,8 @@ def convert(self) -> None:
 				selector = self.dyldFile.readString(targetOff)
 
 				if not selector in self.selectorCache:
-					print("\tNo selref for: " + str(selector))
+					if selector != b'\x00':
```
Comment (arandomdev) [inline DyldExtractor/Converter/SelectorConvertor.py:77]: We should probably figure out why the selector is b"\x00" but this works for now.

**Review by arandomdev — CHANGES_REQUESTED** [review]

**Comment by arandomdev** [comment]
So I noticed that you're using an x86_64 version of the DYLD Cache, and while most of the converters should work fine, StubConverter and SelectorConvertor will either fail or output bad data, because the "instructions" that it uses targets the Arm64 instruction set. For now, we can probably just put up a warning about it but in the long term, it would be nice if we can support x86_64.

Thanks!

**Pull request merged**